app/src/org/commcare/android/resource/installers/XFormAndroidInstaller.java

```diff
@@ -91,21 +91,20 @@ protected int customInstall(Resource r, Reference local, boolean upgrade, Androi
         }
 
         Vector<Integer> existingforms = FormDefRecord.getFormDefIdsByJrFormId(platform.getFormDefStorage(), formDef.getMainInstance().schema);
-        if (existingforms != null && existingforms.size() > 0) {

```

Comment:
We are removing this check to decide whether we wanna create a new form def entry or not and making this decision based on the `upgrade` flag now which conveys whether we have an exisiting resource with the same id in the master table.